test/serializer_test.exs

```diff
@@ -410,6 +453,133 @@ defmodule JSONAPISerializerTest do
     end
   end
 
+  describe "when underscore_to_dash: except" do
+    setup do
+      Application.put_env(:jsonapi, :underscore_to_dash,
+        except: [:full_description, :_private_attribute, :metadata]
+      )
+
+      on_exit(fn ->
+        Application.delete_env(:jsonapi, :underscore_to_dash)
+      end)
+
+      {:ok, []}
+    end
+
+    test "serialize properly uses underscore_to_dash on both attributes and relationships" do
+      data = %{
+        id: 1,
+        text: "Hello",
+        inserted_at: NaiveDateTime.utc_now(),
+        metadata: %{series_title: "Beginnings", __private_code: 123, double__underscore: true},
+        _private_attribute: "please do not touch",
+        nonstandard__attribute: "no rules apply",
+        body: "Hello world",
+        full_description: "This_is_my_description",
+        author: %{id: 2, username: "jbonds", first_name: "jerry", last_name: "bonds"},
+        best_comments: [
+          %{
+            id: 5,
+            text: "greatest comment ever",
+            user: %{id: 4, username: "jack", last_name: "bronds"}
+          }
+        ]
+      }
+
+      encoded = Serializer.serialize(PostWithMetadataView, data, nil)
+
+      attributes = encoded[:data][:attributes]
+      relationships = encoded[:data][:relationships]
+      included = encoded[:included]
+
+      assert attributes["inserted-at"] == data[:inserted_at]
+      assert attributes["nonstandard__attribute"] == data[:nonstandard__attribute]
+
+      # Not modified
+      assert attributes["full_description"] == data[:full_description]
+      assert attributes["metadata"] == data[:metadata]
+      assert attributes["_private_attribute"] == data[:_private_attribute]
+
+      assert Enum.find(included, fn i -> i[:type] == "user" && i[:id] == "2" end)[:attributes][
```

Comment:
I personally find these `assert` statements very difficult to grok, can we clean them up in any way?  It may be possible to use a pipeline:

```elixir
assert "bonds" == included
                  |> Enum.find(&(&1.type == "user" && &1.id == "2")
                  |> get_in([:attributes, "last-name"])

# Or perhaps with a capture 
last_name = included
            |> Enum.find(&(&1.type == "user" && &1.id == "2")
            |> get_in([:attributes, "last-name"])

assert "bonds" == last_name


```

Reply:
I agree! These are modeled after other tests in this file (see [line 441](https://github.com/jeregrine/jsonapi/pull/137/files#diff-e5e7be16724a4eee8799ab05f1d9cc83R441)). I thought it would be better to keep same style while adding tests, to focus on the content of this addition.

Another option would be to add some helper functions to the tests:

```elixir
assert "bonds" ==
  included
  |> find_user_with_id("2")
  |> get_attribute("last-name")
```

Do you want to address this here, or would it be better to do a separate PR to improve readability of these tests?

Reply:
Agree with the pipeline style for these tests.  And I would imagine another PR would be preferable.